Fe 16 add form submit result view

abdevworks opened the pull request:
https://fe-16-add-form-submit-result-view--melo-envelo.netlify.app/theme

Wyświetlony przykład z innym butonem niż na makiecie. Nie ma to tutaj znaczenia, jako że zarówno buton jak i ikona przekazywane są przez content projection.

abdevworks committed:
feat: form-submit-result-view

Refactored component to use content projection

abdevworks committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/MELO-front into FE-16-add-form-submit-result-view

abdevworks committed:
feat: form-submit-result-view

Fixed import in theme.component

abdevworks committed:
feat: form-submit-result-view

Changed background color to gradient to be consistent with the rest of the app

abdevworks committed:
feat: add-form-submit-result-view

Fixed bg color styling so its using global styling class

dominikapap commented on src/app/shared/ui/form-result-view/form-result-view.component.ts:
```diff
@@ -0,0 +1,34 @@
+};
+
+@Component({
+  selector: 'app-form-result-view',
```
Czy tutaj nie byłoby może lepiej zmienić selector `app-form-result-view` na `app-form-result-view[formResultInfo]` żeby mieć pewność, że Input będzie przekazany?

abdevworks replied:
Zastanawiałem się dokładnie nad takim rozwiązaniem. Pytanie tylko co dla ciebie będzie wyglądało lepiej. To, że będziesz miała error, że nie można znaleźć takiego selectora jak ktoś nie poda Inputa i będzie się zastanawiał, gdzie popełnił błąd, czy bez tego i wtedy jeśli nie poda Inputa to wyświetli się mu pusty komunikat bez żadnego tekstu.

abdevworks replied:
Dodałem zmiany zgodnie z twoją sugestią. Komponent można będzie utworzyć tylko jeśli został podany input [formResultInfo].

abdevworks committed:
feat: add-form-submit-result-view

Changed component selector so it cant be created without [formResultInfo] input information

dominikapap commented on src/app/core/theme.component.ts:
```diff
@@ -1,8 +1,21 @@
+    <app-form-result-view [formResultInfo]="formResultInfo">
+      <img src="../../../assets/form-result-icons/confirm-icon.svg" alt="" />
+      <button>-></button>
+    </app-form-result-view>
```
Tu możnaby jeszcze dodać jakiś nagłówek czym jest ten komponent jest bo będziemy mieli dużo elementów w theme to będzie się łatwiej odnaleźć jeśli będą jakoś oddzielone albo zatytułowane

akademiaEnvelo commented on src/app/shared/ui/form-result-view/form-result-view-component.scss:
```diff
@@ -0,0 +1,13 @@
+.frv-container {
```
`frv` nic mi nie mówi

abdevworks replied:
Nazwa poprawiona na 'form-result-view-container'

akademiaEnvelo commented on src/app/shared/ui/form-result-view/form-result-view.component.ts:
```diff
@@ -0,0 +1,34 @@
+@Component({
+  selector: 'app-form-result-view[formResultInfo]',
+  standalone: true,
+  imports: [CommonModule],
```
zamiast całego commonModule możemy zaimportować oddzielnie `NgIf`

abdevworks replied:
Import zmieniony na _import { NgIf } from '@angular/common';_

abdevworks committed:
feat: add-submit-result-view

Added header in theme. Changed class name to be more readable.Fixed NgIf imprt

akademiaEnvelo commented on src/app/app.module.ts:
```diff
@@ -11,6 +11,7 @@ import { environment } from "src/environment";
 import { RouterModule } from "@angular/router";
 import { noProductionGuard } from "@shared/no-production.guard";
 
+
```
jakąs zabłkąkana dodatkowa linijka się pojawiła

abdevworks replied:
Zbędna linia została usunięta.

abdevworks committed:
feat: add-form-submit-result-view

Removed unnecessary line in app.module

abdevworks committed:
feat: searchbar

Updated with data pulled from develop

rafalfuz commented on src/app/features/event/event.state.service.ts:
```diff
@@ -0,0 +1,49 @@
+})
+export class EventStateService {
+  private eventState$$ = new BehaviorSubject<EventState>(eventStateDefault);
+  constructor(private http: HttpClient) {}
```
Moge się mylic dlatego tylko pytanie z mojej strony, czy nie powinienes użyć Inject() do wstrzykniecie http, wiem ze działa tak samo jak w użyciu w constructorze ale kojarzę z wykładów, że powinniśmy się przyzwyczajać właśnie do zapisu bez uzycia constructora

abdevworks replied:
Sugestia dobra, tylko odnosi się do taska z event serwisem także u siebie zostawiam tak jak jest, jako że nie jest to błąd.

abdevworks committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/MELO-front into FE-16-add-form-submit-result-view

abdevworks committed:
feat: form-result-view

Fixed form result view

abdevworks committed:
feat: form-submit-result

Added missin changes to component projection
